### PR TITLE
Make Schnorr EdDSA Compatible

### LINKED ISSUE #131: [Sign] Schnorr should be EdDSA compatible for verification
There are three things that makes the current `sign.Schnorr` signature incompatible with the verification function of `eddsa.Verify`. 
+ The `sign.SchnorrSign` uses the hash function of the suite. While in principle it's correct, here that breaks compatibility with the `EdDSA` implementation which uses a `sha512` while the regular `ed25519` suite use `sha256`.
We should switch to `sha512` so our signatures are verifiable by the regular EdDSA verification implementations.

The signature format in `sign.Schnorr` is `c, s` with `c = H(Random || Public  || message)` and `s = r - c*x`. In `eddsa`, the signature is  `R || s`.
+ The schnorr format should follow eddsa format. No need to follow Wikipedia format, since nobody uses it.
+ The signature must be changed from `s = r - c * x` to `s = r + c * x`.

### PR DESCRIPTION
**Pull request by christophebertrand**
closes #131

### CODE REVIEW

**Comment by ineiti** [comment]
Nice work. Some questions for @nikkolasg :

- is this now compatible with https://tools.ietf.org/html/rfc8032?
- will it break code that we have?
- should this go into crypto.v0? or only crypto.v1?

**Comment by nikkolasg** [comment]
@ineiti :
* yes i believe as they kept backwards compatibility.
* Good question: if all of our code is running on `crypto.v0` then we should be fine if we don't merge it with v0. Otherwise, it will break for future signatures (like the dashboard in our office !)
* As mentioned above, I suggest we keep it in master and merge it in v1 when the times come to avoid any un-necessary problems.

File: sign/schnorr_test.go [hunk 1]
```diff
@@ -27,19 +27,38 @@ func TestSchnorrSignature(t *testing.T) {
 	larger := append(s, []byte{0x01, 0x02}...)
 	assert.Error(t, VerifySchnorr(suite, kp.Public, msg, larger))
 
+	constantBytes := []byte{243, 45, 180, 140, 73, 23, 41, 212, 250, 87, 157, 243,
```
Comment (nikkolasg) [inline sign/schnorr_test.go]: where does this comes from .. ?

**Comment by nikkolasg** [comment]
Almost ready to merge, well done !

File: sign/schnorr_test.go (hunk 1 above)
Comment (christophebertrand) [inline sign/schnorr_test.go]: Those bytes are not a valid encoding of Scalar and Point

Comment (ineiti) [inline sign/schnorr_test.go]: Oh, I see. just move the `// wrong challenge` line before the `constantBytes`, then we'll understand it, too ;)

**Pull request merged**